### DISCUSSION TITLE
is it possile to pass a list to a query field?

### DISCUSSION THREAD

**Discussion by Danelfcf** [msiemens/tinydb#495] [Q&A]
Hi I am trying to use tinydb for a parametic search, however I am having a hard time with the Query. what I want to do is the following:
I have a database created like db.insert({'name': 'John', 'age': 22, 'job'='student'}) and I want to setup a search where I only look for the fields name and age. while I can do this by hand I want to be able to pass a list of feilds and a list of values like the following:
```python
def find(self, field, x):
        '''
        Example usage: print(db.find(['type'], ['apple']))
        :param field: list(str) database 'columns' to look at
        :param x: list(str) values in cols to look for
        :return: list[dic{}]
        '''
        return self.db.search(self.q[tuple(field)].one_of(x))

# so in my case I want to input something like:
print(class.find(["Jill", "John"], [22, 23, 85]))
```
while I know how to handle checking for the values using one_of, I cannot find the equivalent command for the 'field'.

**Comment by msiemens** [comment]
Hey @Danelfcf,

I'm not sure what you mean to do with the `self.q[tuple(field)]` call. According to your example code, `field` is a list of strings. And if I understand correctly, you want to return all documents where at least one of the fields includes one of the values passed in `x`. Is that correct? If so, I got something to work like this:


```python
from tinydb import TinyDB, Query

db = TinyDB('test.json')

db.truncate()
db.insert({'name': 'John', 'age': 22, 'job': 'student'})
db.insert({'name': 'John1', 'age': 23, 'job': 'student'})
db.insert({'name': 'John2', 'age': 24, 'job': 'student'})
db.insert({'name': 'John3', 'age': 25, 'job': 'student'})

def find(fields, x):
    '''
    Example usage: print(db.find(['type'], ['apple']))
    :param fields: list(str) database 'columns' to look at
    :param x: list(str) values in cols to look for
    :return: list[dic{}]
    '''
    query = Query()[fields.pop(0)].one_of(x)

    for field in fields:
        query = query | Query()[field].one_of(x)

    return db.search(query)

print(find(['age', 'name'], [22, 23, 'John3']))
```
